tests/Matcher/Pattern/Expander/RepeatTest.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace Coduo\PHPMatcher\Tests\Matcher\Pattern\Expander;
+
+use Coduo\PHPMatcher\Matcher\Pattern\Expander\Repeat;
+
+class RepeatTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @dataProvider examplesProvider
+     */
+    public function test_matching_values($needle, $haystack, $expectedResult, $isStrict = true)
+    {
+        $expander = new Repeat($needle, $isStrict);
+        $this->assertEquals($expectedResult, $expander->match($haystack));
+    }
+
+    public static function examplesProvider()
+    {
+        $jsonPattern = '{"name": "@string@", "activated": "@boolean@"}';
+
+        $jsonTest = array(
```

Comment:
You could use "new" PHP short array syntax `[]` which is less verbose

Reply:
Yes probably, but the whole project seems to use the older `array()` syntax. In order to stick with the project's coding convention I'd rather not change that for now.

Reply:
On master branch we are using `[]` - 2.3 branch was only created for 5.6 support however I'm not planning to add more new features here without adding them to the master branch first